src/libraries/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -639,6 +639,43 @@ public void AssemblyResolve_IsNotCalledForCoreLibResources()
             }).Dispose();
         }
 
+        class CorrectlyPropagatesException : Exception
+        {
+            public CorrectlyPropagatesException(string message) : base(message)
+            { }
+        }
+
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/43909", TestRuntimes.Mono)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void AssemblyResolve_ExceptionPropagatesCorrectly(bool throwOnError)
+        {
+            RemoteExecutor.Invoke((throwOnErrorString) => {
```

Comment:
I would prefer to not use remote executor - it's not necessary here (and it makes the test very slow).
While the event handler is a global thing, we can easily unregister after the test - and the test intentionally tries to load nonexistent assembly, so it will not modify the loaded assemblies in any way.